dlrover/python/diagnose/inferencechain/inferenceoperator/check_failure_node_operator.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2024 The DLRover Authors. All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import List
+
+from dlrover.python.common.log import default_logger as logger
+from dlrover.python.diagnose.common.constants import InferenceConfigKey
+from dlrover.python.diagnose.common.inference_chain import (
+    Inference,
+    InferenceAttribute,
+    InferenceDescription,
+    InferenceName,
+    InferenceOperator,
+)
+from dlrover.python.diagnose.datacollector.training_log_collector import (
+    TrainingLogCollector,
+)
+
+
+class CheckFailureNodeOperator(InferenceOperator):
+    def __init__(self):
+        super().__init__()
+
+    def is_compatible(self, inference: Inference) -> bool:
+        if (
+            inference.name == InferenceName.NODE
+            and inference.attribution == InferenceAttribute.ISORNOT
+            and inference.description == InferenceDescription.FAILURE
+        ):
+            return True
+        else:
+            return False
+
+    def infer(self, inferences: List[Inference]) -> List[Inference]:
+        if (
+            len(inferences) == 0
+            or not inferences[0].configs
+            or InferenceConfigKey.LOG_FILE not in inferences[0].configs
+            or InferenceConfigKey.ERRORS not in inferences[0].configs
```

Comment:
We can't implement like 'inferences[0]' here. Need to provide a function to search specified inference by name.